packages/tadviewer/src/PivotRequester.ts

```diff
@@ -372,9 +375,12 @@ export class PivotRequester {
               (vs: ViewState) =>
                 vs
                   .update("loadingTimer", (lt) => lt.stop())
-                  .set("viewParams", prevViewParams) as ViewState
```

Comment:
I'm a bit reluctant to give this up.
The basic idea is that we don't want to leave the component in a bad state when an error occurs, we'd like to restore it to the "last known good" state.  
Here is a thought:  Right now we're `prevViewParams`, which is really just the last set `pendingViewParams`.   As you've observed, the problem seems to be that `pendingViewParams` might have itself been bad, causing us to just restore to a state that causes us to run a bad query over and over again.
What if, instead, we added a new property, `lastKnownGoodViewParams` that has type `ViewParams | null`.  We start it off as `null`, and then set it to `viewParams` in the `.then()` block since at that point we know the query succeeded.
Then, in this error handling case, we call `.set("viewParams", lastKnownGoodViewParams ?? vs.viewParams)`.  wdyt?

Reply:
The last parameters are good (with the exception of the initial null scenario). The bad piece is the baseQuery, for which we don't keep a good version. I am going to introduce a `prevBaseQuery` and see if that works to fix the situation.

Reply:
oooooh, I see.  Yeah, if it's the `baseQuery` itself that's messed up, I can definitely see why no amount of resetting `viewParams` will help (and, as you've observed, actually makes things worse, by repeatedly running the broken query...).
While keeping track of `prevBaseQuery` is an admirable idea, it could get pretty involved, because you'd also need to keep track of the corresponding `schema` and `viewParams` and ensure that they are all restored consistently in the case of error.  That might be worth doing, but it's pretty involved, and we should probably only go down that route if we can do some kind of fault injection and write a reasonable assertive test, which gets to be a much larger, more complex undertaking.
So, on second thought:  let's land this with your proposed change here (removing line 375) as-is.  Please just tighten up the typing on `err` from `any` to `Error` mentioned in earlier comment and I will approve.